File: codewars/k5/luck_check.py
import math


def luck_check(st):
    """Checks whether the sums of both halves of a string equal each
    other. Odd-length strings have the middle digit disregarded.
    """
    notIntegers = not math.prod([x.isdigit() for x in st])

    if len(st) == 0 or notIntegers:
        raise Exception
    
    if len(st) % 2:
        print(bool(len(st) % 2))
        st = st[0 : math.floor(len(st)/2)] + st[math.ceil(len(st)/2) :]

    leftSum = sum([int(x) for x in st[0 : int(len(st)/2)]])
    fullSum = sum([int(x) for x in st])

    return True if leftSum == (fullSum / 2) else False 

File: challenges/codewars/javascript/k5/luck-check.js
'use strict';


/**
 * Checks whether the sums of both halves of a string equal each other.
 * Odd-length strings have the middle digit disregarded.
 */
function luckCheck(ticket) {
  const l = ticket.length;
  const notIntegers = ![...ticket].every(x => '1234567890'.includes(x));

  if (ticket.length === 0 || notIntegers) {
    throw new Error();
  }

  if (ticket.length % 2) {
    ticket = ticket.slice(0, Math.floor((l/2))) 
    + ticket.slice(Math.ceil((l/2), l));
  }

  const halfSum = [...ticket.slice(0, (l/2))].reduce((a, c) => a + +c, 0);
  const fullSum = [...ticket].reduce((a, c) => a + +c, 0);

  return halfSum === (fullSum / 2) ? true : false;
}